relayer/workers/parachaincommitmentrelayer/beefy-listener.go

```diff
@@ -77,7 +77,15 @@ func (li *BeefyListener) Start(ctx context.Context, eg *errgroup.Group) error {
 	li.beefyLightClient = beefyLightClientContract
 
 	eg.Go(func() error {
-		err := li.subBeefyJustifications(ctx)
+
+		blockNumber, hash, err := li.fetchLatestBlockAndHash()
+		if err != nil {
+			return nil
+		}
+
+		li.catchupMissedCommitments(ctx, blockNumber, hash)
+
+		err = li.subBeefyJustifications(ctx)
```

Comment:
Catchup could take longer than one justification, so we could miss additional commitments. We might need to change the catchup logic to run within every justification we receive.

Reply:
yeh planning to implement in this PR.